web_programming/crawl_google_results.py

```diff
@@ -19,4 +19,5 @@
 
     print(len(links))
     for link in links:
-        webbrowser.open(f"http://google.com{link.get('href')}")
+        while (link.text == "Maps"): webbrowser.open(f"{link.get('href')}")[:0]
```

Comment:
So many things confuse me about this single line of code!
1. What search term would show us why this change is useful?
2. Why use `while` instead of `if`?  Won’t that send us in an infinite loop?
3. Why put the parentheses around `link.text == "Maps"`?  psf/black will remove these.
4. Why put the conditional and the statement on the same line?  psf/black will put them on separate lines.
5. Why create an f-string that contains no {}?
6. `webbrowser.open()` will always return `None` so why try to slice `None`?
7. `>>> 'abc'[:0]` is the empty string ('').  Why try to create an empty slice and then not use it?

Please explain.

Reply:
Thanks for your feedback.  Your criticisms are well founded.

Reply:
1. The search term used was "bitcoin"
2. You are correct, I misused the while statement.
3. You are correct.
4. You are correct.
5. Each statement has f strings.
6. To solve the while problem I created for myself, inadvertently.
7. You are correct.